### ISSUE TITLE
Count fails for small series

### ISSUE THREAD

**Issue by denis-jasselette-jc** [iamlikeme/rainflow#54]
Hello,

While testing an application of mine, I innocently tried to count cycles on very small vectors (size < 3). To my surprise, it yielded a few different obscure exceptions depending on the size of the series. I figured it was not supported and I even understand why, as no results would really make sense in those edge cases. So I just checked the size of my vector to guard from these exceptions.

However, the exceptions were not very helpful about the source of the problem and I couldn't find any documentation stating that there was such a limit on the input. So maybe, it could be documented somewhere, or maybe we could change slightly the reversals function with these edge cases in mind...

Here is a quick illustration of what I am suggesting:

```python
import unittest
import rainflow

def _reversals(series):
    """Iterate reversal points in the series.
    A reversal point is a point in the series at which the first derivative
    changes sign. Reversal is undefined at the first (last) point because the
    derivative before (after) this point is undefined. The first and the last
    points are treated as reversals.
    Parameters
    ----------
    series : iterable sequence of numbers
    Yields
    ------
    Reversal points as tuples (index, value).
    """
    series = iter(series)

    x_last, x = next(series, None), next(series, None)
    if x_last is None or x is None:
        return
        
    d_last = (x - x_last)

    yield 0, x_last
    index = None
    for index, x_next in enumerate(series, start=1):
        if x_next == x:
            continue
        d_next = x_next - x
        if d_last * d_next < 0:
            yield index, x
        x_last, x = x, x_next
        d_last = d_next

    if index is not None:
        yield index + 1, x_next

rainflow.reversals = _reversals

class TestRainflow(unittest.TestCase):
    def test_reversals_small_series(self):
        assert list(rainflow.reversals([])) == []
        assert list(rainflow.reversals([1])) == []
        assert list(rainflow.reversals([1, 2])) == [(0, 1)]
        assert list(rainflow.reversals([1, 2, 3])) == [(0, 1), (2, 3)]

    def test_extract_cycles_small_series(self):
        assert list(rainflow.extract_cycles([])) == []
        assert list(rainflow.extract_cycles([1])) == []
        assert list(rainflow.extract_cycles([1, 2])) == []
        assert list(rainflow.extract_cycles([1, 2, 3])) == [(2, 2.0, 0.5, 0, 2)]
```

The result of `extract_cycles` or `count_cycles` would be an empty list, which makes some sense at least.

Any thoughts?

**Comment by iamlikeme** [comment]
This looks very elegant. I will be glad to merge a pull request if you submit one :smile:

**Comment by iamlikeme** [comment]
Implemented in #56.
